hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/RowCounter.java

```diff
@@ -65,22 +68,42 @@ public class RowCounter extends AbstractHBaseTool {
   private final static String OPT_END_TIME = "endtime";
   private final static String OPT_RANGE = "range";
   private final static String OPT_EXPECTED_COUNT = "expectedCount";
+  private final static String OPT_COUNT_DELETE_MARKERS = "countDeleteMarkers";
 
   private String tableName;
   private List<MultiRowRangeFilter.RowRange> rowRangeList;
   private long startTime;
   private long endTime;
   private long expectedCount;
+  private boolean countDeleteMarkers;
   private List<String> columns = new ArrayList<>();
 
+  private Job job;
```

Comment:
this doesnot look necessary, we have been validating counters with following logic for existing tests. Please update tests to take a similar approach:
```
   * Run the RowCounter map reduce job and verify the row count.
   * @param args          the command line arguments to be used for rowcounter job.
   * @param expectedCount the expected row count (result of map reduce job).
   * @throws Exception in case of any unexpected error.
   */
  private void runCreateSubmittableJobWithArgs(String[] args, int expectedCount) throws Exception {
    Job job = RowCounter.createSubmittableJob(TEST_UTIL.getConfiguration(), args);
    long start = EnvironmentEdgeManager.currentTime();
    job.waitForCompletion(true);
    long duration = EnvironmentEdgeManager.currentTime() - start;
    LOG.debug("row count duration (ms): " + duration);
    assertTrue(job.isSuccessful());
    Counter counter = job.getCounters().findCounter(RowCounter.RowCounterMapper.Counters.ROWS);
    assertEquals(expectedCount, counter.getValue());
  }
```

Reply:
Although this might become a little messy as we are adding test asserts in real code, not sure if what exists is right way.!

Reply:
@NihalJain, the method `runCreateSubmittableJobWithArgs` internally calls  `RowCounter.createSubmittableJob(TEST_UTIL.getConfiguration(), args)`. However, the method `createSubmittableJob` is marked for deprecation - [code link](https://github.com/apache/hbase/blob/40d9e9db776203cdb6ccf2dacebe7503c68bd79c/hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/RowCounter.java#L142). So ideally, I believe (please correct me if I am wrong), we should not be making a change to that method. To use that method, we have to change the scan behaviour based on the flag.

Reply:
Hey @shubham-roy I mean we could rewriten a helper in tests similar to above example  method `runCreateSubmittableJobWithArgs` and make assertions. IMO we should try to get rid of deprecated API as another task than mixing implementations and doing same thing in different ways at different places.

Reply:
>IMO we should try to get rid of deprecated API as another task than mixing implementations and doing same thing in different ways at different places.

@NihalJain , don't you think that the access to the job object via a getter method (which I exposed) could be a good starting point to getting rid of the deprecated method `createSubmittableJob`. I already used it in a way that could be easily extended to other use cases as well. LMK what do you think.

Reply:
Sure but I would prefer to do that as cleanup task for separation of concerns

Reply:
>Sure but I would prefer to do that as another cleanup task for separation of concerns

@NihalJain , I agree and I am also not touching any of the other tests. I just used whatever is needed for my testing in an extensible way. Fixing of remaining tests can be taken up as a separate cleanup task.

Reply:
will leave this upto others as i am still mot convinced. +0 from me.

Reply:
@virajjasani , can you please have a look at this thread and let us know of your thoughts on the same?